src/main/groovy/org/shipkit/internal/notes/contributors/ContributorsDispatcher.java

```diff
@@ -0,0 +1,45 @@
+package org.shipkit.internal.notes.contributors;
+
+
+import org.json.simple.JsonObject;
+import org.shipkit.internal.notes.model.ProjectContributor;
+import org.shipkit.internal.notes.util.Function;
+import org.shipkit.internal.notes.util.GitHubObjectFetcher;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class ContributorsDispatcher {
+
+    private static final int N_THREADS = 4;
+
+    public Set<ProjectContributor> dispatch(List<JsonObject> page, String readOnlyAuthToken) {
+        Set<ProjectContributor> result = new HashSet<ProjectContributor>();
+        ExecutorService executor = Executors.newFixedThreadPool(N_THREADS);
+        List<JsonObject> synchronizedPageList = Collections.synchronizedList(page);
+
+        List<Future<Set<ProjectContributor>>> futures = new ArrayList<Future<Set<ProjectContributor>>>();
+        if (page.size() > 0) {
+            for (int i = 0; i < N_THREADS; i++) {
+                GitHubObjectFetcher objectFetcher = new GitHubObjectFetcher(readOnlyAuthToken);
```

Comment:
This method has sufficient meat that is worth unit testing :)

I was thinking how to make it neatly unit testable (btw. my favorite thing about engineering is figure out how I can cut code so that it is easy to test :). fetcher and fetcherFunction are immutable so you can pass fetcherFunction as parameter to "dispatch" method. This should make "dispatch" method unit testable easily.